docs/DirectManipulation.md

```diff
@@ -0,0 +1,224 @@
+any of its children will have their opacity updated in response to taps.
+
+```javascript
+<TouchableOpacity onPress={this._handlePress}>
+  <View style={styles.button}>
+    <Text>Press me!</Text>
+  </View>
+<TouchableOpacity>
+```
+
+Let's imagine that `setNativeProps` was not available. One way that we
+might implement it with that constraint is to store the opacity value
+in the state, then update that value whenever `onPress` is fired:
+
+```javascript
+getInitialState() {
+  return { myButtonOpacity: 1, }
+},
+
+render() {
+  return (
+    <TouchableOpacity onPress={() => this.setState({myButtonOpacity: 0.5})}
+                      onPressOut={() => this.setState({myButtonOpacity: 1})}>
+      <View style={[styles.button, {opacity: this.state.myButtonOpacity}]}>
+        <Text>Press me!</Text>
+      </View>
+    </TouchableOpacity>
+  )
+}
+```
+
+This is a lot of work compared to the original example - you need to
+track the opacity value and re-render the entire component each time the
+opacity changes. It's possible that you could simplify this further, but
+it will ultimately be some variation of the same approach.
+
+If you look at the implementation of `setNativeProps` in
+[NativeMethodsMixin.js](https://github.com/facebook/react-native/blob/master/Libraries/ReactIOS/NativeMethodsMixin.js)
+you will notice that it is a wrapper around `RCTUIManager.updateView` -
+this is the exact same function call that results from re-rendering -
+see [receiveComponent in
+ReactNativeBaseComponent.js](https://github.com/facebook/react-native/blob/master/Libraries/ReactNative/ReactNativeBaseComponent.js).
+
+## Composite components and setNativeProps
+
+Composite components are not backed by a native view, so you cannot call
+`setNativeProps` on them. Consider this example:
+
+```javascript
+var MyButton = React.createClass({
+  render() {
+    return (
+      <View>
+        <Text>{this.props.label}</Text>
+      </View>
+    )
+  },
+});
+
+var App = React.createClass({
+  render() {
+    return (
+      <TouchableOpacity>
+        <MyButton label="Press me!" />
+      </TouchableOpacity>
+    )
+  },
+});
+```
+
+If you run this you will immediately see this error: `Touchable child must either be native or forward setNativeProps to a native component`. This occurs because `MyButton` actually has no representation in the visible UI - there is no `UIView` for it, and you can't set the opacity of something that doesn't exist. You can think about it like this: if you define a component with `React.createClass` you would not expect to be able to set a style prop on it and have that work - you would need to pass the style prop down to a child, unless you are wrapping a native component. So the two options we have, as the error message so kindly suggests, are to make the composite component into a native component or to forward `setNativeProps` to child component with native backing (in the same way that we would pass style down to a child).
```

Comment:
If you run this you will immediately see this error: `Touchable child must either be native or forward setNativeProps to a native component`. This occurs because `MyButton` isn't directly backed by a native view whose opacity should be set. You can think about it like this: if you define a component with `React.createClass` you would not expect to be able to set a style prop on it and have that work - you would need to pass the style prop down to a child, unless you are wrapping a native component. So the two options we have, as the error message so kindly suggests, are to make the composite component into a native component or to forward `setNativeProps` to child component with native backing (in the same way that we would pass style down to a child).